backend-shared/hack/service_account.go

```diff
@@ -0,0 +1,170 @@
+	if err != nil {
+		if !apierr.IsAlreadyExists(err) {
+			return fmt.Errorf("Failed to create service account %q in namespace %q: %v", serviceAccountName, namespace, err)
+		}
+		return fmt.Errorf("ServiceAccount %q already exists in namespace %q", serviceAccountName, namespace)
+	}
+	log.Printf("ServiceAccount %q created in namespace %q", serviceAccountName, namespace)
+	return nil
+}
+
+func InstallServiceAccount(clientset kubernetes.Interface, ns string, namespaces []string) (string, error) {
+	// ArgoCDManagerServiceAccount = "argocd-manager", is the name of the service account for managing a cluster
+
+	err := CreateServiceAccount(clientset, "argocd-manager", ns)
+	if err != nil {
+		return "", err
+	}
+	if len(namespaces) != 0 {
+		for _, namespace := range namespaces {
+			role := rbacv1.Role{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "rbac.authorization.k8s.io/v1",
+					Kind:       "Role",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ArgoCDManagerClusterRole,
+				},
+				Rules: ArgoCDManagerNamespacePolicyRules,
+			}
+			_, err := clientset.RbacV1().Roles(namespace).Create(context.Background(), &role, metav1.CreateOptions{})
+			if err != nil {
+				if !apierr.IsAlreadyExists(err) {
+					_, err := clientset.RbacV1().Roles(namespace).Update(context.Background(), &role, metav1.UpdateOptions{})
+					if err != nil {
+						return "", err
+					}
+				}
+			}
+
+			roleBinding := rbacv1.RoleBinding{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "rbac.authorization.k8s.io/v1",
+					Kind:       "RoleBinding",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ArgoCDManagerClusterRoleBinding,
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "Role",
+					Name:     ArgoCDManagerClusterRole,
+				},
+				Subjects: []rbacv1.Subject{{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      ArgoCDManagerServiceAccount,
+					Namespace: ns,
+				}},
+			}
+			_, err = clientset.RbacV1().RoleBindings(namespace).Create(context.Background(), &roleBinding, metav1.CreateOptions{})
+			if err != nil {
+				if !apierr.IsAlreadyExists(err) {
+					_, err := clientset.RbacV1().RoleBindings(namespace).Update(context.Background(), &roleBinding, metav1.UpdateOptions{})
+					if err != nil {
+						return "", err
+					}
+				}
+			}
+		}
+	}
+	return GetServiceAccountBearerToken(clientset, ns, ArgoCDManagerServiceAccount)
+}
+
+// GetServiceAccountBearerToken will attempt to get the provided service account until it
+// exists, iterate the secrets associated with it looking for one of type
+// kubernetes.io/service-account-token, and return it's token if found.
+func GetServiceAccountBearerToken(clientset kubernetes.Interface, ns string, sa string) (string, error) {
+	var serviceAccount *corev1.ServiceAccount
+	var secret *corev1.Secret
+	var err error
+	err = wait.Poll(500*time.Millisecond, 30*time.Second, func() (bool, error) {
```

Comment:
`500` is a magic number. A magic number is a numeric literal that is not defined as a constant, but which may change, and therefore can be hard to update later. If you think we are going to re-use this in other function as well, please create a constant.

Reply:
TIL, thanks! I don't think we might need it, I'll update it!